src/erlcloud_kinesis.erl

```diff
@@ -406,6 +407,73 @@ put_record(StreamName, PartitionKey, Data, ExplicitHashKey, Ordering, Config) wh
   Json = [{<<"StreamName">>, StreamName}, {<<"PartitionKey">>, PartitionKey}, {<<"Data">>, base64:encode(Data)}, {<<"ExplicitHashKey">>, ExplicitHashKey}, {<<"SequenceNumberForOrdering">>, Ordering}],
   erlcloud_kinesis_impl:request(Config, "Kinesis_20131202.PutRecord", Json).
 
+%%------------------------------------------------------------------------------
+%% Kinesis API:
+%% [https://docs.aws.amazon.com/kinesis/latest/APIReference/API_PutRecords.html]
+%%
+%% ===Example===
+%%
+%% This operation puts data records into an Amazon Kinesis stream from a producer.
+%%
+%% `
+%% erlcloud_kinesis:put_records(<<"test">>, [{<<"asdasd">>, <<"asdasd">>}]).
+%% {ok, [{<<"FailedRecordCount">>, 0},
+%%      {<<"Records">>,
+%%       [
+%%        [{<<"SequenceNumber">>,
+%%          <<"49537292605574028653758531131893428543501381406818304001">>},
+%%         {<<"ShardId">>,<<"shardId-000000000000">>}]
+%%       ]}
+%%     ]
+%% }
+%% erlcloud_kinesis:put_records(<<"test">>, [{<<"asdasd">>, <<"asdasd">>, <<"asdasd">>}]).
+%% {ok, [{<<"FailedRecordCount">>, 0},
+%%      {<<"Records">>,
+%%       [
+%%        [{<<"SequenceNumber">>,
+%%          <<"49537292605574028653758531131893428543501381406818304001">>},
+%%         {<<"ShardId">>,<<"shardId-000000000000">>}]
+%%       ]}
+%%     ]
+%% }
+%% '
+%%
+%% @end
+%%------------------------------------------------------------------------------
+
+-spec put_records(string(), Items) -> proplist() when
+      Items :: [Item],
+      Item :: {string(), string()}
```

Comment:
My issue with the example is maybe a microcosm of the fact that you have to read the code for prepare_put_records_data, etc. to understand the form of the input. It's not clear what the order is supposed to be here—what element of the tuple is data, hash key, partition key?

Might be more helpful to make types, although I guess it's not done in general in this module. e.g.:

``` erlang
-type put_records_item() :: {Data :: string(),  PartitionKey :: string()} |
                            {Data :: string(), ExplicitHashKey :: string(), PartitionKey :: string()}.
-type put_records_items() :: [put_records_item()].
```

Then you can just have

```erlang
spec put_records(string(), put_records_items()) -> proplist().
...
spec put_records(string(), put_records_items(), aws_config()) -> proplist().
```

seems more succinct and easy to read.

As an aside, I notice that your spec calls for strings, but your examples are all binary—which are *not* strings. I am also not 100% sure that it would even work with strings. It seems like this might be a larger problem with the authorship of this module... but shouldn't all these instances of `string()` in the specs actually be `binary()`?